python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -1356,12 +1357,81 @@ def get() -> "OpExecutionContext":
         ctx = _current_asset_execution_context.get()
         if ctx is None:
             raise DagsterInvariantViolationError("No current OpExecutionContext in scope.")
-        return ctx.get_op_execution_context()
+        return ctx.op_execution_context
+
+
+###############################
+######## AssetExecutionContext
+###############################
+
+ALTERNATE_METHODS = {
+    "run_id": "run_properties.run_id",
+    "run": "run_properties.dagster_run",
+    "dagster_run": "run_properties.dagster_run",
+    "run_config": "run_properties.run_config",
+    "retry_number": "run_properties.retry_number",
+}
+
+
+def _get_deprecation_kwargs(attr: str):
+    deprecation_kwargs = {"breaking_version": "1.8.0"}
+    deprecation_kwargs["subject"] = f"AssetExecutionContext.{attr}"
+
+    if attr in ALTERNATE_METHODS:
+        deprecation_kwargs["additional_warn_text"] = (
+            f"You have called the deprecated method {attr} on AssetExecutionContext. Use"
+            f" context.{ALTERNATE_METHODS[attr]} instead."
+        )
+
+    return deprecation_kwargs
+
+
+class RunProperties(
+    NamedTuple(
+        "_RunProperties",
+        [
+            ("run_id", PublicAttr[str]),
+            ("dagster_run", PublicAttr[DagsterRun]),
+            ("run_config", PublicAttr[Mapping[str, object]]),
+            ("retry_number", PublicAttr[int]),
+        ],
+    )
+):
+    """Relevant information about a run while it is executing. Contains:
+    * The run ID
+    * The DagsterRun object, which contains further information about the run
+    * The run config
+    * The retry number - how many times this step has been retried.
+    """
+
+    def __new__(
+        cls,
+        run_id: str,
+        dagster_run: DagsterRun,
+        run_config: Mapping[str, object],
+        retry_number: int,
+    ):
+        return super(RunProperties, cls).__new__(
+            cls,
+            run_id=run_id,
+            dagster_run=dagster_run,
+            run_config=run_config,
+            retry_number=retry_number,
+        )
```

Comment:
if theres nothing meaningful happening in `__new__` you can do the nicer terse `NamedTuple` declaration (I assume PublicAttr still works with that)